darshan-util/darshan-posix-logutils.c

```diff
@@ -587,6 +661,7 @@ static void darshan_log_agg_posix_files(void *rec, void *agg_rec, int init_flag)
                 break;
             case POSIX_STRIDE1_STRIDE:
             case POSIX_ACCESS1_ACCESS:
+                memset(duplicate_mask, 0, 4*sizeof(duplicate_mask[0]));
```

Comment:
See comment added in 0cdf2375..0ed17804.  The reason we have to explicitly reset the duplicate_mask[] is because the exact same logic is used twice in the loop; once for strides and once for access sizes.  It needed a comment to point that out; I just added it.

And that's right, the only reason that array is on the stack now at all is because the previous logic kept up with state by modifying counters in place as it went.  That's not legal any more since someone will likely still want to use raw records after passing them through an accumulator.

Reply:
Ahh, whoops, you're right. Thanks for adding the comment in any case!